operator/pkg/utils/iamauthenticator/awsiamauthenticator.go

```diff
@@ -0,0 +1,215 @@
+		Spec: appsv1.DaemonSetSpec{
+			UpdateStrategy: appsv1.DaemonSetUpdateStrategy{Type: appsv1.RollingUpdateDaemonSetStrategyType},
+			Selector: &metav1.LabelSelector{
+				MatchLabels: authenticatorLabels(),
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{Labels: authenticatorLabels()},
+				Spec: PodSpec(obj, func(spec v1.PodSpec) v1.PodSpec {
+					spec.NodeSelector = nodeSelector
+					spec.Volumes = append(spec.Volumes, v1.Volume{
+						Name: "config",
+						VolumeSource: v1.VolumeSource{
+							ConfigMap: &v1.ConfigMapVolumeSource{
+								LocalObjectReference: v1.LocalObjectReference{Name: AuthenticatorConfigMapName(obj.GetName())},
+							},
+						},
+					})
+					return spec
+				}),
+			},
+		},
+	}))
+}
+
+type Options func(v1.PodSpec) v1.PodSpec
+
+func PodSpec(obj client.Object, opts ...Options) v1.PodSpec {
+	spec := podSpec(obj)
+	for _, opt := range opts {
+		spec = opt(spec)
+	}
+	return spec
+}
+
+func podSpec(obj client.Object) v1.PodSpec {
+	return v1.PodSpec{
+		HostNetwork: true,
+		Tolerations: []v1.Toleration{{Operator: v1.TolerationOpExists}},
+		InitContainers: []v1.Container{{
+			Name:  "chown",
+			Image: imageprovider.BusyBox(),
+			Command: []string{"sh", "-c",
+				"touch /var/aws-iam-authenticator/kubeconfig/kubeconfig.yaml && chown -R 10000:10000 /var/aws-iam-authenticator/state/ && chown -R 10000:10000 /var/aws-iam-authenticator/kubeconfig && ls -lrt /var/aws-iam-authenticator",
+			},
+			SecurityContext: &v1.SecurityContext{AllowPrivilegeEscalation: ptr.Bool(true)},
+			VolumeMounts: []v1.VolumeMount{{
+				Name:      "state",
+				MountPath: "/var/aws-iam-authenticator/state/",
+			}, {
+				Name:      "kubeconfig",
+				MountPath: "/var/aws-iam-authenticator/kubeconfig/",
+			}},
+		}},
+		Containers: []v1.Container{{
+			Name:  "aws-iam-authenticator",
+			Image: imageprovider.AWSIamAuthenticator(),
+			Args: []string{
+				"server",
+				"--master=https://localhost/",
+				"--config=/etc/aws-iam-authenticator/config.yaml",
+				"--state-dir=/var/aws-iam-authenticator/state/",
+				"--generate-kubeconfig=/var/aws-iam-authenticator/kubeconfig/kubeconfig.yaml",
+			},
+			SecurityContext: &v1.SecurityContext{AllowPrivilegeEscalation: ptr.Bool(true)},
+			VolumeMounts: []v1.VolumeMount{{
+				Name:      "config",
+				MountPath: "/etc/aws-iam-authenticator/",
+			}, {
+				Name:      "state",
+				MountPath: "/var/aws-iam-authenticator/state/",
+			}, {
+				Name:      "kubeconfig",
+				MountPath: "/var/aws-iam-authenticator/kubeconfig/",
+			}},
+		}},
+		Volumes: []v1.Volume{{
+			Name: "kubeconfig",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/var/aws-iam-authenticator/kubeconfig/",
+				},
+			},
+		}, {
+			Name: "state",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/var/aws-iam-authenticator/state/",
+				},
+			},
+		}},
+	}
+}
+
+func authenticatorLabels() map[string]string {
```

Comment:
If this is static, why is it a function? Can we autogenerate these somehow, assuming all components generically have the label "component"